test.js

```diff
@@ -619,4 +619,25 @@ if ('DOMException' in globalThis) {
 		const serialized = serializeError(new DOMException('x'));
 		t.is(serialized.message, 'x');
 	});
+
+	test('should deep clone DOMException when it is in the cause property', t => {
+		const domException = new DOMException('My domException', 'NotFoundError');
+		const error = new Error('My error message', {
+			cause: domException,
+		});
+
+		const serialized = serializeError(error);
+
+		t.is(serialized.message, 'My error message');
+		t.is(serialized.cause.message, 'My domException');
+		t.is(serialized.cause.name, 'NotFoundError');
+		t.is(serialized.cause.code, 8);
+		// Should preserve DOMException constants
+		t.is(serialized.cause.NOT_FOUND_ERR, 8);
+		t.is(serialized.cause.SYNTAX_ERR, 12);
```

Comment:
This sounds weird, why are these constants serialized? Also I don't see any overlap with the request in #107 

It's a static prop:

<img width="227" height="42" alt="Screenshot" src="https://github.com/user-attachments/assets/266bb59f-c0ef-40cb-bd52-c11ea3385de1" />

Reply:
Yeah, I turned the PR into just being about adding a test.